### PR TITLE
Add VM migration walkthrough

### PR DESCRIPTION
**Pull request by gthvn1**
As I'm currently trying to understand how a VM is migrated by XAPI I think it could be interesting to keep my journey as a walk through. Every comments are welcome so don't hesitate. I'm not an English native speaker so feel free to propose any improvements. And of course any technical inaccuracies as well.

I hope that this walk trough will be useful,
Thanks in advance for comments

### CODE REVIEW

File: ocaml/xenopsd/doc/walk-throughs/VM.migrate.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
+Walk-through: Migrating a VM
+============================
+
+# From client to Xenopsd
+
+A Xenopsd client wishes to migrate a VM from one host to another within
```
Comment (psafont) [inline ocaml/xenopsd/doc/walk-throughs/VM.migrate.md:6]: Do you mean a xapi client? They way to control VMs from outside the host is through xapi, not xenopsd.

Comment (gthvn1) [reply ocaml/xenopsd/doc/walk-throughs/VM.migrate.md:6]: yes absolutely, I mean xapi client.

File: ocaml/xenopsd/doc/walk-throughs/VM.migrate.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+- VM.rename
+- VBD.set_active
+- VBD.plug
+- VIF.set_active
+- VGPU.set_active
+- VM.create_device_model
+- PCI.plug
+- VM.set_domain_action_request
+
+Several backends are implementend but in this document we will assume that we use Xen via libxc, libxenguest
```
Comment (psafont) [inline ocaml/xenopsd/doc/walk-throughs/VM.migrate.md:51]: A single backend exists now: libxc

Comment (gthvn1) [reply ocaml/xenopsd/doc/walk-throughs/VM.migrate.md]: fixed

File: ocaml/xenopsd/doc/walk-throughs/VM.migrate.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+more information about the generated functions you can have a look to
+[XAPI IDL model](https://github.com/xapi-project/xen-api/tree/master/ocaml/idl/ocaml_backend).
+
+The command have serveral parameters such as should it be ran asynchronously,
+should it be forwared to another host, how arguments should be marshalled and
+so on. A new thread is created by [xapi/server_helpers.ml](https://github.com/xapi-project/xen-api/blob/7ac88b90e762065c5ebb94a8ea61c61bdbf62c5c/ocaml/xapi/server_helpers.ml#L55)
+to handle the command asynchronously. At this point the helper also check if
+the command should be passed to the [message forwarding](https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/message_forwarding.ml)
+layer in order to be executed on another host (the destination) or locally if
+we are already at the right place.
```
[the remark this answers is not in the document]
Comment (gthvn1) [reply ocaml/xenopsd/doc/walk-throughs/VM.migrate.md:35]: ok sure. If I understand correctly I moved the xapi_vm_migrate operation where you proposed.

**Comment by robhoes** [comment]
Hi @gthvn1, thanks for the document!

We are in the process of moving all documentation into a new site based on Hugo, with all files now under the top-level `doc` directory. It is not yet complete, but I've just raised a PR to move the xenopsd docs into the new location: #5273. I've also added a sequence chart for live migration, which I had created a few weeks ago. The docs will be visible here: https://xapi-project.github.io/new-docs.

I would suggest to wait for my PR to get merged, and then move your new document into the new location.

**Comment by gthvn1** [comment]
@robhoes yes sure I will wait until your PR is merged and I will move it to the new location. I will follow that :+1:

**Comment by robhoes** [comment]
@gthvn1 It's merged now. Could you move your doc?

**Comment by gthvn1** [comment]
@robhoes yes sure I will do that quickly

**Comment by gthvn1** [comment]
nit: I also notice that `_index.md` is using **Walk-Throughs** or **Walk-throughs** and I copied what has been done in *VM.start* that is **Walkthrough**. If needed I can modify files for consistency. All of them are fine for me :)

**Pull request merged**